src/models/transformer.py

```diff
@@ -14,22 +14,36 @@
 
 class BackboneModel(TrainableModel):
 
-    def __init__(self, backbone: nn.Module, x_dim: int, n_positions: int, n_embd: int=128, y_dim: int = 1):
+    def __init__(self, 
+                 backbone: nn.Module, 
+                 x_dim: int, 
+                 n_positions: int, 
+                 n_embd: int=128, 
+                 y_dim: int = 1,
+                 interleave: bool = True,):
         super().__init__(x_dim, y_dim)
 
         self.context_length = n_positions
         self._read_in = nn.Linear(x_dim, n_embd)
         self._backbone = backbone
         self._read_out = nn.Linear(n_embd, y_dim)
 
+        self._interleave = interleave
+
     def forward(self, xs: Tensor, ys: Tensor):
         self._backbone.to(xs.device) # pyright: ignore[reportArgumentType,reportAttributeAccessIssue]
 
-        zs = self.interleave(xs, ys)
+        zs = xs
+        if self._interleave:
+            zs = self.interleave(xs, ys)
```

Comment:
I discussed this with @KesavaViswanadha , we might be able to eliminate this entirely (making this PR much slimmer lol)

Since the main reason to not interleave x/y pairs is for MQAR, we should try to change the implementation to support this.

Reply:
how would we get away with eliminating the need to not interleave for MQAR? It feels like this would break MQAR if we didn't change anything else